### PR TITLE
Add support for function values in AsInsert and AsUpdate.

### PR DESCRIPTION
**Pull request by dimarobert**
+ NULLIF function implemented.

### CODE REVIEW

File: QueryBuilder/Compilers/Compiler.cs [hunk 1]
```diff
@@ -777,8 +777,18 @@ public virtual string WrapValue(string value)
             return opening + value.Replace(closing, closing + closing) + closing;
         }
 
-        public virtual string Parameter<T>(SqlResult ctx, T value)
+        public virtual string Parameter<T>(SqlResult ctx, T value) 
         {
+            if (value is AbstractFunctionValue)
+            {
+                if (value is NullifValue)
```
Comment (CMarius94) [inline QueryBuilder/Compilers/Compiler.cs:784]: change to ->  if (value is NullifValue **nullifValue**)

File: QueryBuilder/Compilers/Compiler.cs [hunk 2]
```diff
@@ -777,8 +777,18 @@ public virtual string WrapValue(string value)
...
+                {
+                    var nullifValue = value as NullifValue;
```
Comment (CMarius94) [inline QueryBuilder/Compilers/Compiler.cs:786]: and you can remove this now

File: QueryBuilder/Values/NullifValue.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿namespace SqlKata.Values {
+    public class NullifValue : AbstractFunctionValue {
+
+        public object Expression => Arguments[0];
+
+        public object Condition => Arguments[1];
```
Comment (CMarius94) [inline QueryBuilder/Values/NullifValue.cs]: this is not a condition... it's a second expression actually.
if it equals to first, then the first value is returned. Let's change its name to Operand or ReferenceValue or something like that

**Review by CMarius94 — CHANGES_REQUESTED** [review]

File: QueryBuilder/Compilers/Compiler.cs (hunk 1 above)
Comment (dimarobert) [reply QueryBuilder/Compilers/Compiler.cs:784]: C# 7.1 support is not enabled in the project.

File: QueryBuilder/Compilers/Compiler.cs (hunk 2 above)
Comment (dimarobert) [reply QueryBuilder/Compilers/Compiler.cs:786]: see above.

**Review by CMarius94 — APPROVED** [review]

**Pull request merged**